### PR TITLE
Feature/post3tos update listcontroller create to really create a list

### PR DESCRIPTION
**Pull request by hannahbit**
Huh, that one needed a lot of looking things up :) But I guess I learned a lot.

### CODE REVIEW

File: lib/do_it_web/controllers/list_controller.ex [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
 
-    render(conn, "show.json", list: list)
+    case result do
+      {:ok, list}         -> conn
+                             |> put_status(200)
+                             |> render("show.json", list: list)
+      {:error, changeset} -> conn
+                             |> put_status(400)
+                             |> render("error.json", changeset: changeset)
+    end
```
Comment (agatheblues) [inline lib/do_it_web/controllers/list_controller.ex:24]: Nice 🎉 ! You can use `mix format` to format your code in a "normal" way 💅

File: lib/do_it_web/views/list_view.ex [hunk 2]
```diff
@@ -8,4 +8,15 @@ defmodule DoItWeb.ListView do
...
+  def render("error.json", %{changeset: changeset}) do
+
+    errors = Ecto.Changeset.traverse_errors(changeset, fn {msg, opts} ->
+      Enum.reduce(opts, msg, fn {key, value}, acc ->
+        String.replace(acc, "%{#{key}}", to_string(value))
+      end)
+    end)
+
+    %{errors: errors}
+  end
```
Comment (agatheblues) [inline lib/do_it_web/views/list_view.ex:21]: Feel free to ignore if you want to move on! ⛸️ 

It is likely you will need the same error-management logic for the todos view. An improvement you could do here is to move this into a custom `ErrorView`. Then you will need in `lib/do_it_web/controllers/list_controller.ex` to tell phoenix "hey my `error.json` is in another view". Hint: https://hexdocs.pm/phoenix/Phoenix.Controller.html?#put_view/2

**Review by agatheblues — APPROVED** [review]
Bravo 👏 There was a lot to tame in this ticket!

File: lib/do_it_web/controllers/list_controller.ex (hunk 1 above)
Comment (hannahbit) [reply lib/do_it_web/controllers/list_controller.ex:29]: did it :)

**Pull request merged**